content/articles/building-a-simple-translation-app-using-python-for-beginners/index.md

```diff
@@ -0,0 +1,171 @@
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /building-a-simple-translation-app-using-python-for-beginners/hero.png
+    alt: translation app example image 
+---
+
+Want to build your very own language translation app?
+
+<!--more-->
+
+Well, in this tutorial, I'll show you how you can build one yourself using Hugging Face Transformers and Gradio!
+
+### Prerequisites
+To understand the contents of this tutorial, a reader needs to be familiar with the Python programming language and machine learning. You can use either Google Colab or Jupyter notebook. In this tutorial, I'm going to be using [Google Colab](https://colab.research.google.com) notebook. Lastly, you need to be familiar with [Natural Language Processing](https://www.section.io/engineering-education/five-real-life-use-cases-of-natural-language-processing-nlp/) and [PyTorch](https://www.section.io/engineering-education/tensorflow-vs-pytorch/).
+
+### Table of contents
+- [Introduction](#introduction)
+- [Installing dependencies](#installing-dependencies)
+- [Downloading the pre-trained translation model](#downloading-the-pre-trained-translation-model)
+- [Building the web app for interaction using Gradio](#building-the-web-app-for-interaction-using-gradio)
+- [Wrapping up](#wrapping-up)
+
+### Introduction
+The queries you make on websites such as Google and smart assistance such as Google assistant all have one thing in common, they both use Natural Language Process (NLP). NLP is a cutting-edge AI technology that gives a machine the ability to read, understand, and derive meaning from human languages. NLP is the reason why Google assistant sounds so intelligent and Google queries are responded to so intelligently. Data analysts and machine learning experts use NLP be mimic human linguistics behavior saving manpower and time. Every day, you use NLP unknowingly in your phone's auto-correct feature helping you correctly spell a word. To understand how NLP works in-depth, please refer to this [article](https://www.section.io/engineering-education/five-real-life-use-cases-of-natural-language-processing-nlp/).
+
+NLP is a huge library that can be difficult to understand for beginners. Because of this, easy-to-use libraries such as [Hugging Face](https://www.section.io/engineering-education/hugging-face/) have been created to democratize NLP. It is one of the leading startups in the NLP space. This tutorial aims to use it to build a simple translation app.
+
+Hugging Face is a great open-source library that is doing powerful work in the Natural Language Processing (NLP) space. The library has a bunch of pre-trained models which you can leverage or fine-tune. The library has many models including BERT and GPT-2 models that perform various tasks, but for our case, we'll be leveraging the pre-trained language pipeline. Rather than going ahead and training a huge language model such as GPT, one can leverage the ML pipeline instead.
```

Comment:
Please add one external resource each for BERT, GPT-2 and GPT, and edit "but for our case" to "but for our purpose"